### ISSUE TITLE
Unrepairable unit damage after upgrading from 0.46.1 to 0.47.15

### ISSUE THREAD

**Issue by BowtieBob** [MegaMek/mekhq#2299]
I loaded a campaign from 0.46.1 into 0.47.15, and all of my dropships and jumpships are showing that they are damaged, but they are not in the repair bay, and I cannot repair them.  I have tried using the GM tools to restore them, but it does nothing.  I also have two Puma Assault Tanks that are showing up in the repair bay, but I can't fix them either.  They both show that SRM ammo bins need to be reloaded.  No techs are showing that will let me do the repair, and if I use the GM tools to complete the task, it changes to Flamer bins needing reloaded.  If I GM complete that task, the SRM bins come back, and it just alternates between them if I keep trying.

### Environment
What version of MekHQ does your issue apply to?  0.47.15
What operating system are you using?  Windows
What java version are you using? Java 8 Update 271

![image](https://user-images.githubusercontent.com/70545594/102700330-12f7b400-4212-11eb-972f-7468ac523869.png)
![image](https://user-images.githubusercontent.com/70545594/102700338-1ee37600-4212-11eb-8e0a-df21a7a3dacb.png)
[Error.zip](https://github.com/MegaMek/mekhq/files/5720197/Error.zip)

**Comment by sixlettervariables** [comment]
Dropships and JumpShips have a known issue with bay doors.

Sounds like the Puma's are out of sync with their entities. Can you attach your megameklog.txt and MekHQlog.txt?

**Comment by BowtieBob** [comment]
There are log files in the attached zip, along with my campaign file and custom units

**Comment by sixlettervariables** [comment]
Those Pumas have an issue between the unit and the entity. It appears that either (a) a refit went bad, or (b) the unit was fixed between then.
```

15:53:46,570 WARN [mekhq.Utilities] {SwingWorker-pool-1-thread-1} 
unscrambleEquipmentNumbers(), line 1309 : Could not unscramble equipment for Puma Assault Tank PAT-005b (1a5b9f0c-26c3-4cfc-a17a-a2c017e8f218)

 - Streak SRM 2 Ammo Bin equipmentNum: 12

Equipment Parts:
 0: Medium Laser Front Medium Laser 
 1: Medium Laser Front Medium Laser 
 2: Streak SRM 2 Front Streak SRM 2 
 3: Streak SRM 2 Front Streak SRM 2 
 4: LRM 20 Right LRM 20 
 5: Artemis IV FCS Right Artemis IV FCS 
 6: LRM 20 Left LRM 20 
 7: Artemis IV FCS Left Artemis IV FCS 
 8: Small Pulse Laser Rear Small Pulse Laser 
 9: Flamer (Vehicle) Rear Flamer (Vehicle) 
 10: ER PPC Turret ER PPC 
 12: Streak SRM 2 Ammo Bin Body <Incorrect>  (Missing)
 12: Vehicle Flamer Ammo Bin Body Vehicle Flamer Ammo 
 14: LRM 20 Artemis-capable Ammo Bin Body LRM 20 Artemis-capable Ammo 
 15: LRM 20 Artemis-capable Ammo Bin Body LRM 20 Artemis-capable Ammo 
 16: LRM 20 Artemis-capable Ammo Bin Body LRM 20 Artemis-capable Ammo 
 17: LRM 20 Artemis-capable Ammo Bin Body LRM 20 Artemis-capable Ammo 

Equipment:
 0: Medium Laser Medium Laser
 1: Medium Laser Medium Laser
 2: Streak SRM 2 Streak SRM 2
 3: Streak SRM 2 Streak SRM 2
 4: LRM 20 LRM 20
 5: Artemis IV FCS Artemis IV FCS
 6: LRM 20 LRM 20
 7: Artemis IV FCS Artemis IV FCS
 8: Small Pulse Laser Small Pulse Laser
 9: Flamer (Vehicle) Flamer (Vehicle)
 10: ER PPC ER PPC
 11: CASE CASE (Available)
 12: Vehicle Flamer Ammo Vehicle Flamer Ammo
 13: LRM 20 Artemis-capable Ammo LRM 20 Artemis-capable Ammo (Available)
 14: LRM 20 Artemis-capable Ammo LRM 20 Artemis-capable Ammo
 15: LRM 20 Artemis-capable Ammo LRM 20 Artemis-capable Ammo
 16: LRM 20 Artemis-capable Ammo LRM 20 Artemis-capable Ammo
 17: LRM 20 Artemis-capable Ammo LRM 20 Artemis-capable Ammo
```
There is no SRM bin available, hence it changes to flamer (due to not figuring it out properly).

I suggest GM Adding two new Pumas, moving the crews, then GM removing the old ones.

**Comment by sixlettervariables** [comment]
With the dropships, it appears somewhere in the conversion from 0.46.1 to 0.47.15 they dropped their bays to 0 tons. I'm not sure how. A new dropship on 0.47.15 doesn't seem to have this problem.

**Comment by BowtieBob** [comment]
I was able to remove all of the units and generate new ones.  It seems to be working now.

**Comment by sixlettervariables** [comment]
I'm going to investigate the bay problem some more.

**Comment by sixlettervariables** [comment]
Do you have a copy of that Campaign from 0.46.1?

**Comment by BowtieBob** [comment]
The campaign in the zip file that was attached to the original post is from 0.46.1

**Comment by sixlettervariables** [comment]
Thank you! It appears that when we read the campaign in 0.47.15 the weapon bay comes back with a size of 0 tons, even though the ammo has more than 0 tons. Still not sure of the direct cause.

**Comment by sixlettervariables** [comment]
Okay, figured this out: when cleaning up ammo parts older campaigns stored the size of LargeCraftAmmoBin's in `<capacity>` rather than `<size>` as in other equipment parts. In consolidating the XML reading and writing logic I missed this difference in XML element name.

**Comment by sixlettervariables** [comment]
This should be a fixed CPNX you can use in 0.47.15:
[NightstalkersBaseline.cpnx.zip](https://github.com/MegaMek/mekhq/files/5721582/NightstalkersBaseline.cpnx.zip)
